test/e2e/fixtures/when.go

```diff
@@ -62,6 +85,14 @@ func (w *When) WaitForWorkflow(timeout time.Duration) *When {
 	}
 }
 
+func (w *When) Wait(timeout time.Duration) *When {
+	logCtx := log.WithFields(log.Fields{"test": w.t.Name(), "cron workflow": w.cronWorkflowName})
+	logCtx.Infof("Waiting for %s", humanize.Duration(timeout))
```

Comment:
would it be possible for this to wait until a condition is met and the exit (unless it timesout) - that way test might be faster

Reply:
I intended this to be a general `wait` function, I think it could potentially be useful as is in the future.

W.r.t. the cron tests, see https://github.com/argoproj/argo/pull/1758#discussion_r363346602